erpnext/quality_management/doctype/quality_procedure/quality_procedure.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018, Frappe and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe.utils.nestedset import NestedSet
+
+class QualityProcedure(NestedSet):
+	nsm_parent_field = 'parent_quality_procedure'
+
+	def before_save(self):
+		for data in self.procedure_step:
+			if data.procedure == 'Procedure' and data.procedure_name:
+				data.step = data.procedure_name
+				self.is_group = 1
+
+	def on_update(self):
+		self.set_parent()
+
+	def after_insert(self):
+		self.set_parent()
+
+	def on_trash(self):
+		if self.parent_quality_procedure:
+			doc = frappe.get_doc("Quality Procedure", self.parent_quality_procedure)
+			for data in doc.procedure_step:
+				if data.procedure_name == self.name:
+					doc.procedure_step.remove(data)
+					doc.save()
+			flag_is_group = 0
+			doc.load_from_db()
+			for data in doc.procedure_step:
+				if data.procedure == "Procedure":
+					flag_is_group = 1
+			if flag_is_group == 0:
+				doc.is_group = 0
+				doc.save()
+
+	def set_parent(self):
```

Comment:
there must be a validation that the child procedure is not used in another procedure (since this is a nested set)

Reply:
Added a validation to check if it is another procedures child